staging/src/k8s.io/api/authentication/v1/types.go

```diff
@@ -105,3 +106,64 @@ type ExtraValue []string
 func (t ExtraValue) String() string {
 	return fmt.Sprintf("%v", []string(t))
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// TokenRequest requests a token for a given service account.
+type TokenRequest struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec TokenRequestSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	// +optional
+	Status TokenRequestStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// TokenRequestSpec contains client provided parameters of a token request.
+type TokenRequestSpec struct {
+	// Audiences are the intendend audiences of the token. A recipient of a
+	// token must identitfy themself with an identifier in the list of
+	// audiences of the token, and otherwise should reject the token. A
+	// token issued for multiple audiences may be used to authenticate
+	// against any of the audiences listed but implies a high degree of
+	// trust between the target audiences.
+	Audiences []string `json:"audiences" protobuf:"bytes,1,rep,name=audiences"`
+
+	// ValidityDuration is the requested duration of validity of the request. The
+	// token issuer may return a token with a different validity duration so a
+	// client needs to check the 'expiration' field in a response.
+	// +optional
+	ValidityDuration *int64 `json:"validityDuration" protobuf:"bytes,2,opt,name=validityDuration"`
```

Comment:
ExpirationSeconds, update the godoc to describe it, and change the status to also be ExpirationSeconds.  Also, the godoc on status is also wrong.

Reply:
One is a duration and one is a time. It's a bit confusing to have both with the same name.

Reply:
If it's a time, then it should be a timestamp, and `ExpirationTimestamp metav1.Time` (DeletionTimestamp).  If it's the seconds remaining before expiration, then it should be `ExpirationSeconds` (like DeletionGracePeriodSeconds, I guess, although that is not calculated).  I'm a bit confused why we would make this "seconds remaining before it expires" (which would require it to be virtual).  

If it's really a time, I lean towards `ExpirationTimestamp` in status, in which case this should be `ExpirationSeconds` in spec (the number of seconds I want this token to last before expiration).

Reply:
I would expect a user to request a duration of time for the token to be valid, (e.g. thirty minutes) and that the status return the time that the token will expire (e.g. unix epoch time 1517854484).